security/monitoring.adoc

```diff
@@ -0,0 +1,197 @@
+[[security-monitoring]]
+= Monitoring cluster events and logs
+{product-author}
+{product-version]
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[security-monitoring-intro]]
+== Introduction
+
+In addition to security measures mentioned in other sections of this
+guide, the ability to monitor and audit an {product-title} cluster is an
+important part of safeguarding the cluster itself and its users against
+inappropriate usage.
+
+There are two main sources of cluster-level information that
+are useful for this purpose: events and logs.
+
+[[security-monitoring-events]]
+== Cluster events
+
+Cluster administrators are encouraged to familiarize themselves with the _Event_ resource
+type and review a xref:../dev_guide/events.adoc#events-reference[list of events] to
+determine which may be of interest. There are typically more potential
+event types than listed here, depending on master controller and plugin
+configuration.
+
+Events are associated with a namespace, either the namespace of the
+resource they are related to or, for cluster events, the *_default_*
+namespace. The *_default_* namespace is likely to hold the most relevant
+events for monitoring or auditing a cluster, both *_Node_* events as well
+as resource events related to infrastructure components like the registry
+and router.
+
+The master API and `*oc*` command do not at this time provide parameters
+to scope a listing of events to only those related to nodes. A simple
+approach would be to use grep:
+
+----
+$ oc get event -n default | grep Node
+1h         20h         3         origin-node-1.example.local   Node      Normal    NodeHasDiskPressure   ...
+----
+
+A more flexible approach is to output the events in a form that other
+tools can then process. For example, the following example uses the `*jq*`
+tool against JSON output to extract only *_NodeHasDiskPressure_* events:
+
+----
+$ oc get events -n default -o json \
+  | jq '.items[] | select(.involvedObject.kind == "Node" and .reason == "NodeHasDiskPressure")'
+
+{
+  "apiVersion": "v1",
+  "count": 3,
+  "involvedObject": {
+    "kind": "Node",
+    "name": "origin-node-1.example.local",
+    "uid": "origin-node-1.example.local"
+  },
+  "kind": "Event",
+  "reason": "NodeHasDiskPressure",
+  ...
+}
+----
+
+Events related to resource creation, modification, or deletion globally
+may also be good candidates for detecting misuse of the cluster. As an
+example, to look for excessive pulling of images, the following query
+counts the number of such recent events:
+
+----
+$ oc get events --all-namespaces -o json \
+  | jq '[.items[] | select(.involvedObject.kind == "Pod" and .reason == "Pulling")] | length'
+
+4
+----
+
+Note that when a namespace is deleted, its events are deleted along with
```

Comment:
Maybe we can put this info inside a `NOTE` admonition block.

Reply:
Fixed.